jstests/ssl/x509/mkcert.py

```diff
@@ -520,17 +601,43 @@ def process_ecdsa_leaf(cert):
     reqargs = ['openssl', 'req', '-new', '-key', key, '-out', csr, '-subj', subject]
     x509args = ['openssl', 'x509', '-in', csr, '-out', pem, '-req', '-CA', ca, '-CAkey', ca, '-days', '7300', '-sha256', '-set_serial', serial]
     if mode == 'server':
-        reqargs.append('-reqexts=v3_req')
+        reqargs = reqargs + ['-reqexts', 'v3_req']
         extfile = tempfile.mkstemp()[1]
-        open(extfile, 'wt').write('subjectAltName=DNS:localhost,DNS:127.0.0.1')
+        open(extfile, 'wt').write('basicConstraints=CA:FALSE\n')
+        open(extfile, 'at').write('subjectAltName=DNS:localhost,IP:127.0.0.1\n')
+        open(extfile, 'at').write('subjectKeyIdentifier=hash\n')
+        key_usage = ('keyUsage=nonRepudiation,digitalSignature,keyEncipherment\n'
+                     if (cert.get('tags') and "responder" in cert['tags'])
+                     else 'keyUsage=digitalSignature,keyEncipherment\n')
+        open(extfile, 'at').write(key_usage)
+        extended_key_usage = ('extendedKeyUsage=serverAuth,clientAuth,OCSPSigning\n'
+                              if (cert.get('tags') and "responder" in cert['tags'])
+                              else 'extendedKeyUsage=serverAuth,clientAuth\n')
+        open(extfile, 'at').write(extended_key_usage)
```

Comment:
Instead of the individual `open` calls that do not close the file (and do not guarantee that the data has been written at all or before the next `open` call happens), the writes will be better expressed with a scope:

<pre>
with open(extfile, 'wt') as f:
  f.write(...)
  #  ...
  f.write(...)
</pre>

Reply:
Ah thanks! I've taken your suggestion :)